### PR TITLE
docs(login): escape codeblocks, and add object store state instructions

### PR DESCRIPTION
**Pull request by metral**
- Addresses the feedback in https://github.com/pulumi/docs/pull/1126#discussion_r291636118
- Closes https://github.com/pulumi/pulumi/issues/2809
- Ref: https://github.com/pulumi/pulumi/pull/2455

### CODE REVIEW

File: cmd/login.go [hunk 1]
```diff
@@ -44,27 +44,37 @@ func newLoginCmd() *cobra.Command {
 			"    $ pulumi login\n" +
 			"\n" +
 			"and this command will prompt you for an access token, including a way to launch your web browser to\n" +
-			"easily obtain one. You can script by using PULUMI_ACCESS_TOKEN environment variable.\n" +
+			"easily obtain one. You can script by using `PULUMI_ACCESS_TOKEN` environment variable.\n" +
 			"\n" +
-			"By default, this will log into app.pulumi.com. If you prefer to log into a separate instance\n" +
-			"of the Pulumi service, such as Pulumi Enterprise, specify a <url>. For example, run\n" +
+			"By default, this will log into `app.pulumi.com`. If you prefer to log into a separate instance\n" +
+			"of the Pulumi service, such as Pulumi Enterprise, specify a URL. For example, run\n" +
```
Comment (justinvp) [inline cmd/login.go]: I think `<url>` was referring to the optional argument to the command, but changing to more general `URL` sounds good to me as we show the command with the URL immediately below.

File: cmd/login.go [hunk 2]
```diff
@@ -44,27 +44,37 @@ func newLoginCmd() *cobra.Command {
...
+			"As a shortcut, you may pass --local to use your home directory (this is an alias for `file://~`):\n" +
 			"\n" +
-			"    $ pulumi login --local\n",
+			"    $ pulumi login --local\n" +
+			"\n" +
+			"Additionally, you may leverage supported object storage backends from one of the cloud providers " +
+			"to manage the state independent of the service. For instance,\n" +
+			"\n" +
+			"AWS S3:\n" +
+			"    $ pulumi login s3://my-pulumi-state-bucket\n" +
```
Comment (justinvp) [inline cmd/login.go]: I _think_ we need an extra `"\n" +` above and below each of these new snippets in order for them to render as code snippets in the docs.

Comment (metral) [reply cmd/login.go]: Feedback has been addressed. PTAL

**Comment by stack72** [comment]
Nice work here @metral :)

**Pull request merged**